src/python/foglamp/core/scheduler.py

```diff
@@ -576,8 +587,21 @@ def max_running_tasks(self, value: int)->None:
 
         self._logger.info("Stopped")
 
+        if self._engine is not None:
+            try:
+                self._engine.close()
+            except Exception:
+                self._logger.exception('Unable to close the database connection pool')
+            self._engine = None
```

Comment:
should be in finally?

Reply:
I'll change it but we don't have a good handle of what should happen when non-Exceptions are raised which indicate very bad situations like out of memory.